### PR TITLE
Allow library users to reset tracking data

### PR DESCRIPTION
**Pull request by stkent**
#### Issue Link

#106 

#### Goals

Allow library users to control the `SharedPreferences` instance used to store all tracking information. In particular, this capability can be used to:

- 'forget' all previously-tracked data (useful if you release a major update and would like to elicit feedback from your entire user base again);
- keep debug build tracking data separate from release build tracking data (should not affect end users, but can be useful for avoiding confusion during development and testing!).

Tasks still to complete:
- [x] update README to explain usage
- [x] update CHANGELOG

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,7 +155,7 @@ The convenience method `applyAllDefaultRules` initializes the prompt timing calc
...
 
-- **The user has never previously provided positive feedback.** We strive to constantly improve our apps' functionality and stability. If we do our job right, there's little to be gained by prompting satisfied users for feedback repeatedly. If we decide to significantly overhaul our app (either internally or externally), we will reset the prompt timing calculator to capture feedback from our entire user base again. // TODO: link to a section that explains how to do this; see #106.
+- **The user has never previously provided positive feedback.** We strive to constantly improve our apps' functionality and stability. If we do our job right, there's little to be gained by prompting satisfied users for feedback repeatedly. If we decide to significantly overhaul our app (either internally or externally), we will [reset](#starting-fresh) the prompt timing calculator to capture feedback from our entire user base again.
```
Comment (stkent) [inline README.md]: fix this link

File: README.md [hunk 2]
```diff
@@ -587,6 +592,54 @@ public class ExampleApplication extends Application {
...
+All data tracked by _amplify_ is stored in a single [`SharedPreferences`](http://developer.android.com/reference/android/content/SharedPreferences.html) instance. By default, this instance is named `"AMPLIFY_SHARED_PREFERENCES_NAME"`. However, it is possible to configure _amplify_ to use a different `SharedPreferences` instance during initialization:
+
+```java
+public class ExampleApplication extends Application {
+    
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        
+        final String sharedPrefsName = "my-custom-shared-prefs-name";
```
Comment (stkent) [inline README.md]: Inline this?

**Comment by btkelly** [comment]
This looks good. I'm curious, was there a request for custom shared preferences? We should talk about backing this in a DB, might handle versioning a bit cleaner.

**Comment by btkelly** [comment]
Approved

**Comment by stkent** [comment]
No specific request to keep this focused on shared preferences - let's talk today about alternatives before this gets merged.

**Comment by btkelly** [comment]
OK sounds good

**Pull request merged**